### DISCUSSION TITLE
A real-time subtitle and translation program using whisper and translatepy for any language.

### DISCUSSION THREAD

**Discussion by mldljyh** [openai/whisper#1422] [Show and tell]
[whisper_real_time_translation](https://github.com/mldljyh/whisper_real_time_translation)

I've created a real-time subtitle and translation program using faster-whisper, which is based on whisper, along with translatepy and tkinter. 
There are still unresolved issues, but if you have any suggestions, please feel free to share them.

**Comment by Shibumi497** [comment]
Very interesting!
It is working, but lagging a lot and repeating twice at least the same sentences. Do you also have these issues?
Thank you in advance,

**Reply by mldljyh** [reply]
The issue of falling significantly behind is most likely due to GPU performance. 
The stuttering of sentences is a problem that needs to be addressed. I am aware that it is an issue related to the NLTK model when separating sentences. 
 As a temporary measure, I'm considering adding a parameter to make it work only through a CLI. The subtitle version requires some further thought.
